### PR TITLE
Integrate ibus methods into input sources

### LINKED ISSUE #333: Selecting an engine in the input method view does not activate it
## What Happened
2020-11-03 19 13 42 の画面録画

Selecting an engine in the input method view does not mark it as an active input method engine.

## Expected Behavior
Selecting an engine in the input method view should mark it as an active input method engine, as you can set the selected keyboard layout as the current keyboard layout in the layout view.

## Steps to Reproduce
1. Open System Settings
2. Go to Keyboard→Input Method. Activate ibus-daemon if it's not. Make sure you have more than two input method engines added in the listbox.
3. Change selected row in the listbox. This does not change the currently active input method engine.

### Logs
N/A

## Platform Information
- elementary OS Odin Daily
- The version of `switchboard-plug-keyboard`: `2.4.1+r1247+pkg988~daily~ubuntu6.0.1`

### PR DESCRIPTION
**Pull request by jeremypw**
Fixes #333 

Ibus sources are added to the org.gnome.desktop.input-sources sources setting in the same way as in Ubuntu. This does not have any material effect at the moment but lays the groundwork for being able to manage both xkb layouts and ibus input methods from the wingpanel indicator in the same way as Ubuntu.

With the latest commit to elementary/wingpanel-indicator-keyboard#86, <Alt>Shift switches between both xkb layouts and ibus engines.

As a consequence of these changes, whenever either an xkb layout or ibus input method is selected in their respective tabs, that becomes the currently active input source.

### CODE REVIEW

**Comment by ryonakano** [comment]
Thank you @jeremypw, this fixes the original issue! It would be great if the selection changes when you change the active engine with the shortcut `Ctrl+Space`, as the selection in Layout tab changes when you press `Alt+Shift`.

**Comment by jeremypw** [comment]
There are too many difficulties/bugs/confusions involved with combining hardware keyboards and ibus input methods (including xkb engines) for me to solve I am afraid.  I'll try and find a different solution.  This installed ibus engines are listed in the ibus settings schema anyway.

**Comment by jeremypw** [comment]
Reopening as issue with Gala KeyboardManager identified and a fix to be proposed

**Comment by ryonakano** [comment]
Not sure if this is related to this branch, but when I open the plug a blank xkb entry is added like this:

```
[('xkb', 'jp'), ('xkb', ''), ('ibus', 'mozc-jp')]
```

**Comment by jeremypw** [comment]
@ryonakano Thanks for the review - I missed that regression introducing a blank xkb setting on opening the plug.  Fixing.

File: src/Views/Layout.vala [hunk 1]
```diff
@@ -110,10 +112,12 @@ namespace Pantheon.Keyboard {
 
             advanced_settings = new AdvancedSettings (panels);
 
-            var entry_test = new Gtk.Entry ();
-            entry_test.valign = Gtk.Align.END;
-            entry_test.expand = true;
-            entry_test.placeholder_text = (_("Type to test your layout"));
+            entry_test = new Gtk.Entry () {
+                hexpand = true,
```
Comment (ryonakano) [inline src/Views/Layout.vala:116]:
```suggestion
                vexpand = true,
```

Seems to cause broken layout

File: src/Widgets/InputMethod/AddEnginesPopover.vala [hunk 2]
```diff
@@ -152,4 +141,6 @@ public class Pantheon.Keyboard.InputMethodPage.AddEnginesPopover : Gtk.Popover {
         listbox.select_row (listbox.get_row_at_index (0));
         search_entry.grab_focus ();
     }
+
+
```
Comment (ryonakano) [inline src/Widgets/InputMethod/AddEnginesPopover.vala:145]:
```suggestion
```

Unintended new lines

**Comment by jeremypw** [comment]
@ryonakano Thanks for the review - will fix asap.

**Review by ryonakano — COMMENTED** [review]
Thank you @jeremypw, now it works as expected and fixes the linked issue.

One thing I noticed is that now activating Mozc always uses en_US layout, not my keyboard layout (Japanese 109 keyboard). Due to this problem I can't type words as printed on the keyboard. On the master branch it uses Japanese 109 keyboard as expected, so something in this branch might change which layout to use.

**Comment by jeremypw** [comment]
@ryonakano Thanks for the review.  I think you need to install this Gala fix to get the correct keyboard: https://github.com/elementary/gala/pull/1041 so this PR will have to be blocked until that is merged.

**Review by ryonakano — DISMISSED** [review]
Thank you for your information, I confirmed it works perfectly now with elementary/gala#1041. LGTM but leaving as unmerged until the linked PR is merged.

**Comment by jeremypw** [comment]
The necessary Gala and Greeter fixes have now been merged.  It is probably best that this PR and the corresponding wingpanel-indicator PR be merged at the same time.

**Review by ryonakano — APPROVED** [review]
Thank you very much @jeremypw! It works as expected. :tada:

**Pull request merged**